app/src/main/java/com/duckduckgo/mobile/android/duckduckgo/ui/tabswitcher/TabSwitcherPresenterImpl.java

```diff
@@ -0,0 +1,98 @@
+package com.duckduckgo.mobile.android.duckduckgo.ui.tabswitcher;
+
+import android.support.annotation.NonNull;
+
+import com.duckduckgo.mobile.android.duckduckgo.ui.browser.tab.Tab;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by fgei on 5/29/17.
+ */
+
+public class TabSwitcherPresenterImpl implements TabSwitcherPresenter {
+
+    private TabSwitcherView tabSwitcherView;
+    private List<Tab> tabs;
```

Comment:
Does this belong in a presenter? It looks like your tabs state lives across a few different classes - some here, some in the TabManager etc. Since you're using MVP it would be great to try and capture this in your Model layer. This will also make your life easier later when you want to start persisting the tabs